crates/transaction-pool/src/validate.rs

```diff
@@ -369,7 +369,8 @@ impl<T: PoolTransaction> ValidPoolTransaction<T> {
 
     /// Returns true if this transaction is underpriced compared to the other.
     pub(crate) fn is_underpriced(&self, other: &Self) -> bool {
-        self.transaction.effective_gas_price() <= other.transaction.effective_gas_price()
+        self.transaction.effective_gas_price() * (100 + PRICE_BUMP) / 100 <=
+            other.transaction.effective_gas_price()
```

Comment:
I believe we need to change this a bit so that this matches:

https://github.com/ethereum/go-ethereum/blob/d40a255e973775575d8d16456252f93ac75c09f0/core/txpool/legacypool/list.go#L288-L306

which looks a bit more complex than the existing check

Reply:
@mattsse Done I've mimic their test in order to check if:
- Transaction A max fee per gas is less that Transaction B max fee per gas + a price bump.
- Or Transaction A max priority fee per gas is less that Transaction B max priority fee per gas + a price bump (in this situation, the two max priority fee per gas should not be zero indicating a non-EIP1559 transaction and invalidating the test).
- Or Transaction A effective gas price is less than Transaction B effective gas price.